modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -41,36 +51,65 @@ public PulsarContainer(String pulsarVersion) {
     public PulsarContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DockerImageName.parse("apachepulsar/pulsar"));
-
         withExposedPorts(BROKER_PORT, BROKER_HTTP_PORT);
-        withCommand("/pulsar/bin/pulsar", "standalone", "--no-functions-worker", "-nss");
-        waitingFor(Wait.forHttp(METRICS_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT));
     }
 
     @Override
     protected void configure() {
         super.configure();
-
-        if (functionsWorkerEnabled) {
-            withCommand("/pulsar/bin/pulsar", "standalone");
-            waitingFor(
-                new WaitAllStrategy()
-                    .withStrategy(waitStrategy)
-                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1))
-            );
-        }
+        setupCommandAndEnv();
     }
 
     public PulsarContainer withFunctionsWorker() {
         functionsWorkerEnabled = true;
         return this;
     }
 
+    public PulsarContainer withTransactions() {
+        transactions = true;
+        return this;
+    }
+
+    public PulsarContainer withConfiguration(String name, String value) {
+        return withEnv("PULSAR_PREFIX_" + name, value);
+    }
+
+    public PulsarContainer withConfiguration(Map<String, String> configuration) {
+        configuration.forEach((name, value) -> withEnv("PULSAR_PREFIX_" + name, value));
+        return this;
+    }
+
     public String getPulsarBrokerUrl() {
         return String.format("pulsar://%s:%s", getHost(), getMappedPort(BROKER_PORT));
     }
 
     public String getHttpServiceUrl() {
         return String.format("http://%s:%s", getHost(), getMappedPort(BROKER_HTTP_PORT));
     }
+
+    protected void setupCommandAndEnv() {
+        String standaloneBaseCommand =
+            "/pulsar/bin/apply-config-from-env.py /pulsar/conf/standalone.conf " + "&& bin/pulsar standalone";
+
+        if (!functionsWorkerEnabled) {
+            standaloneBaseCommand += " --no-functions-worker -nss";
+        }
+
+        withCommand("/bin/bash", "-c", standaloneBaseCommand);
+        if (transactions) {
+            withConfiguration("transactionCoordinatorEnabled", "true");
+        }
+
+        if (functionsWorkerEnabled) {
+            waitingFor(
+                new WaitAllStrategy()
+                    .withStrategy(waitStrategy)
+                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1))
+            );
+        } else if (transactions) {
+            waitingFor(Wait.forHttp(TRANSACTION_TOPIC_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT));
```

Comment:
This is needed because the Pulsar client will try to start the TransactionCoordinatorClient and if the topic doesn't exist yet it will fail. 
So if we remove that this code won't work:

```
try (PulsarContainer pulsar = new PulsarContainer(PULSAR_IMAGE).withTransactions();) {
  try (PulsarClient client = PulsarClient.builder().serviceUrl(pulsarBrokerUrl).enableTransaction(true).build();) {
   ....       
  }
}
```

and you'll get 
```
Caused by: org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException: org.apache.pulsar.client.api.PulsarClientException$BrokerMetadataException: {"errorMsg":"Namespace not found","reqId":3642126209754600375, "remote":"localhost/127.0.0.1:54629", "local":"/127.0.0.1:54823"}
	at app//org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException.unwrap(TransactionCoordinatorClientException.java:131)
	at app//org.apache.pulsar.client.api.transaction.TransactionCoordinatorClientException.unwrap(TransactionCoordinatorClientException.java:129)
	at app//org.apache.pulsar.client.impl.transaction.TransactionCoordinatorClientImpl.start(TransactionCoordinatorClientImpl.java:70)
	at app//org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:204)
	... 47 more
```

Reply:
btw I created this to improve the Pulsar client https://github.com/apache/pulsar/issues/15996